diff --git a/CMakeLists.txt b/CMakeLists.txt
index a9daee7..a0936c3 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,7 +16,14 @@
 #  
 #  CMakeLists.txt -- template and derived ESC/I ESC/I-2 protocol commands
 
-project (epsonscan2)
+option(HEADLESS "HEADLESS" OFF)
+
+if(HEADLESS)
+    project (epsonscan2-headless)
+else()
+    project (epsonscan2)
+endif()
+
 set (CMAKE_INSTALL_PREFIX "/usr")
 cmake_minimum_required (VERSION 2.8.12.2)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c884d83..73958e7 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -27,6 +27,8 @@ add_subdirectory(ES2FileFormat/Linux)
 add_subdirectory(Controller)
 add_subdirectory(SaneWrapper/Linux)
 add_subdirectory(Standalone)
-add_subdirectory(DetectAlert)
+if(NOT HEADLESS)
+	add_subdirectory(DetectAlert)
+endif()
 add_subdirectory(ScanSDK)
 
diff --git a/src/Standalone/CMakeLists.txt b/src/Standalone/CMakeLists.txt
index 32ea6ca..79b46d5 100644
--- a/src/Standalone/CMakeLists.txt
+++ b/src/Standalone/CMakeLists.txt
@@ -13,7 +13,7 @@
 #  but WITHOUT ANY WARRANTY; without even the implied warranty of
 #  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 #  Lesser General Public License for more details.
-#  
+#
 #  CMakeLists.txt -- template and derived ESC/I ESC/I-2 protocol commands
 
 PROJECT(es2standalone)
@@ -24,12 +24,18 @@ set(CMAKE_AUTOMOC ON)
 set(CMAKE_AUTOUIC ON)
 set(CMAKE_AUTORCC ON)
 
-find_package(Qt5Core REQUIRED)
-find_package(Qt5Gui REQUIRED)
-find_package(Qt5Widgets REQUIRED)
+if(NOT HEADLESS)
+    find_package(Qt5Core REQUIRED)
+    find_package(Qt5Gui REQUIRED)
+    find_package(Qt5Widgets REQUIRED)
+endif()
 
 add_definitions(-DBOOST_NO_CXX11_RVALUE_REFERENCES=1)
 
+if(HEADLESS)
+    add_definitions(-DNO_QTMODE=1)
+endif()
+
 INCLUDE_DIRECTORIES(
 ../
 ../../
@@ -39,132 +45,146 @@ INCLUDE_DIRECTORIES(
 ../CommonUtility/utils
 )
 
+
 file(GLOB_RECURSE UI_FILES *.ui)
 file(GLOB_RECURSE QRC_FILES icons/*.qrc)
 
-add_executable(es2standalone 
+if(NOT HEADLESS)
+    set(QT_CXX_FILES mainwindow.cpp
+        capitem.cpp
+        documentsource.cpp
+        scanningside.cpp
+        documentsize.cpp
+        imagetype.cpp
+        resolution.cpp
+        rotate.cpp
+        deskewcheckbox.cpp
+        addpagescheckbox.cpp
+        skipblankpagescomb.cpp
+        detectdoublefeedcomb.cpp
+        imageformatcomb.cpp
+        foldercomb.cpp
+        prefixlineedit.cpp
+        noneradiobutton.cpp
+        removebackradiobutton.cpp
+        removebackcombobox.cpp
+        dropoutcombobox.cpp
+        brightnessslider.cpp
+        contrastslider.cpp
+        gammaslider.cpp
+        gammadoublespinbox.cpp
+        thresholdslider.cpp
+        documentsizedialog.cpp
+        widthdoublespinbox.cpp
+        lengthdoublespinbox.cpp
+        detectpapercheckbox.cpp
+        skipblankpagessettingsdialog.cpp
+        skipblankpagesdetlvslider.cpp
+        detectdoublefeedsettingsdialog.cpp
+        paperthicknesscombobox.cpp
+        configurationdialog.cpp
+        unitscombobox.cpp
+        lastusedcheckbox.cpp
+        transferdataaftercompcheckbox.cpp
+        openimgfolderafterscancheckbox.cpp
+        sleeptimerspinbox.cpp
+        acpowercombobox.cpp
+        batterypowercombobox.cpp
+        scanbutton.cpp
+        deviceselectorcombobox.cpp
+        imageformatsettingsdialog.cpp
+        imageformatsettingscombobox.cpp
+        tiffcompressionmonocombobox.cpp
+        networkdeviceselectdialog.cpp
+        waitingdialog.cpp
+        devicenamestackedwidget.cpp
+        deviceselectdialog.cpp
+        initdeviceselectcombobox.cpp
+        devicefinder.cpp
+        ../Platform.cpp
+        detectionareamindoublespinbox.cpp
+        detectionareamaxdoublespinbox.cpp
+        scanningdialog.cpp
+        selectabledetectdoublefeedsettingsdialog.cpp
+        aboutepsonscan2dialog.cpp
+        addpagedialog.cpp
+        horizondoctoolbutton.cpp
+        verticaldoctoolbutton.cpp
+        documenttype.cpp
+        translationstring.cpp
+        deviceconnectiondialog.cpp
+        savedestfolder.cpp
+        custommessagebox.cpp
+        continuousscandialog.cpp
+        automaticfeedingmodedialog.cpp
+        automaticfeedingmodecombobox.cpp
+        documentsourcesettingsdialog.cpp
+        afmtimeoutcombobox.cpp
+        backgroundcolorcombobox.cpp
+        previewwindow.cpp
+        previewbutton.cpp
+        drawingarea.cpp
+        myrubberband.cpp
+        qualitypreviewcheckbox.cpp
+        filenamesettingdialog.cpp
+        adfpaperprotectioncombobox.cpp
+        behaviorwhendoublefeedcombobox.cpp
+        directpoweroncombobox.cpp
+        nonconnectpoweroffcombobox.cpp
+        sensorglassdirtsensitivitycombobox.cpp
+        imagetypesettingsdialog.cpp
+        administratorpasswordinputdialog.cpp
+            administratorpasswordinputforuksecuritydialog.cpp
+        textenhancecombobox.cpp
+        textenhancesettingdialog.cpp
+        addeditpagedialog.cpp
+        imageeditdialog.cpp
+        imageeditscene.cpp
+        imageeditview.cpp
+        rotateimage.cpp
+            removepunchholes.cpp
+        documentsizesettingdialog.cpp
+        blackedgeinscribecheckbox.cpp
+        focuspositionsettingdialog.cpp
+        autocroppinginscribedcheckbox.cpp
+        ${UI_FILES}
+        ${QRC_FILES}
+    )
+else()
+set(QT_CXX_FILES "")
+endif()
+
+add_executable(es2standalone
 	main.cpp
-    mainwindow.cpp 
-    capitem.cpp 
-    documentsource.cpp 
-    scanningside.cpp
-    documentsize.cpp 
-    imagetype.cpp 
-    resolution.cpp 
-    rotate.cpp 
-    deskewcheckbox.cpp 
-    addpagescheckbox.cpp 
-    skipblankpagescomb.cpp 
-    detectdoublefeedcomb.cpp 
-    imageformatcomb.cpp 
-    foldercomb.cpp 
-    prefixlineedit.cpp 
-    noneradiobutton.cpp 
-    removebackradiobutton.cpp 
-    removebackcombobox.cpp 
-    dropoutcombobox.cpp 
-    brightnessslider.cpp 
-    contrastslider.cpp 
-    gammaslider.cpp 
-    gammadoublespinbox.cpp 
-    thresholdslider.cpp 
-    documentsizedialog.cpp 
-    widthdoublespinbox.cpp 
-    lengthdoublespinbox.cpp
-    detectpapercheckbox.cpp 
-    skipblankpagessettingsdialog.cpp 
-    skipblankpagesdetlvslider.cpp 
-    detectdoublefeedsettingsdialog.cpp 
-    paperthicknesscombobox.cpp 
-    configurationdialog.cpp 
-    configuration.cpp 
-    unitscombobox.cpp 
-    lastusedcheckbox.cpp 
-    transferdataaftercompcheckbox.cpp 
-    openimgfolderafterscancheckbox.cpp 
-    sleeptimerspinbox.cpp 
-    acpowercombobox.cpp 
-    batterypowercombobox.cpp 
-    supervisor.cpp 
-    scanbutton.cpp 
-    deviceselectorcombobox.cpp 
-    imageformatsettingsdialog.cpp 
-    imageformatsettingscombobox.cpp 
-    tiffcompressionmonocombobox.cpp 
-    networkdeviceselectdialog.cpp 
-    waitingdialog.cpp 
-    devicenamestackedwidget.cpp 
-    deviceselectdialog.cpp 
-    initdeviceselectcombobox.cpp 
-    devicelist.cpp 
-    devicefinder.cpp 
-    ../Platform.cpp 
-    detectionareamindoublespinbox.cpp 
-    detectionareamaxdoublespinbox.cpp 
-    scanningdialog.cpp 
-    selectabledetectdoublefeedsettingsdialog.cpp 
-    aboutepsonscan2dialog.cpp 
-    addpagedialog.cpp 
-    horizondoctoolbutton.cpp 
-    verticaldoctoolbutton.cpp 
-    lastusedsettings.cpp 
-    commandlinemode.cpp 
-    documenttype.cpp
-    translationstring.cpp 
+    commandlinemode.cpp
+    devicelist.cpp
+    supervisor.cpp
     saveimagepathlist.cpp
-	deviceconnectiondialog.cpp
-	savedestfolder.cpp
-	custommessagebox.cpp
-	continuousscandialog.cpp
-	automaticfeedingmodedialog.cpp
-	automaticfeedingmodecombobox.cpp
-	documentsourcesettingsdialog.cpp
-	afmtimeoutcombobox.cpp
-	backgroundcolorcombobox.cpp
-	previewwindow.cpp
-	previewbutton.cpp
-	drawingarea.cpp
-	myrubberband.cpp
-	qualitypreviewcheckbox.cpp
-	filenamesettingdialog.cpp
-	adfpaperprotectioncombobox.cpp
-	behaviorwhendoublefeedcombobox.cpp
-	directpoweroncombobox.cpp
-	nonconnectpoweroffcombobox.cpp
-	sensorglassdirtsensitivitycombobox.cpp
-	defaultsettings.cpp
-	imagetypesettingsdialog.cpp
-	administratorpasswordinputdialog.cpp
-        administratorpasswordinputforuksecuritydialog.cpp
-	textenhancecombobox.cpp
-	textenhancesettingdialog.cpp
-	addeditpagedialog.cpp
-	imageeditdialog.cpp
-	imageeditscene.cpp
-	imageeditview.cpp
-	rotateimage.cpp
-        removepunchholes.cpp
-    documentsizesettingdialog.cpp
-    blackedgeinscribecheckbox.cpp
-    focuspositionsettingdialog.cpp
-    autocroppinginscribedcheckbox.cpp
-	${UI_FILES}
-   	${QRC_FILES}
+    configuration.cpp
+    lastusedsettings.cpp
+    defaultsettings.cpp
+    ${QT_CXX_FILES}
 )
-
-if (${CMAKE_VERSION} LESS 3.0)
-    qt5_wrap_ui(FORMS ${UI_FILES})
-    add_custom_target(uic_cmake28_compat ALL DEPENDS ${FORMS})
+if (NOT HEADLESS)
+    if (${CMAKE_VERSION} LESS 3.0)
+        qt5_wrap_ui(FORMS ${UI_FILES})
+        add_custom_target(uic_cmake28_compat ALL DEPENDS ${FORMS})
+    endif()
 endif()
 
 set_target_properties(es2standalone  PROPERTIES OUTPUT_NAME "epsonscan2")
 
-target_link_libraries(es2standalone ${QT_LIBRARIES}
-	pthread
+if(NOT HEADLESS)
+    target_link_libraries(es2standalone ${QT_LIBRARIES})
+endif()
+
+target_link_libraries(es2standalone pthread
 	dl	
 	commonutility
 )
 
-QT5_USE_MODULES(es2standalone Widgets)
+if(NOT HEADLESS)
+    QT5_USE_MODULES(es2standalone Widgets)
+endif()
 install(TARGETS es2standalone DESTINATION "${EPSON_INSTALL_ROOT}/usr/bin")
 
